### PR TITLE
ci(smoke): use aws arm64 runner

### LINKED ISSUE #11590: ARM64 CI Jobs frequently fail due to connection resets
Examples:
- https://github.com/camunda/zeebe/actions/runs/4134636764/jobs/7146014786
- https://github.com/camunda/zeebe/actions/runs/4130548513/jobs/7137350957

Could be related to MTU or GCP network interference. See https://github.com/zeebe-io/infra/blob/main/gcp/zeebe-io/zeebe-ci-1/runner-arm64-4.yml for the configuration we use for these runners

### PR DESCRIPTION
**Pull request by megglos**
## Description

Uses the new arm runners provided by infra https://confluence.camunda.com/display/HAN/Github+Actions+Self-Hosted+Runners for the arm smoke test.

Pool can scale up to 20 runners https://dashboard.int.camunda.com/d/izL-_Qlnz/gha-self-hosted-runners?orgId=1&var-runnerlabel=aws-arm-core-4-unstable

10+ runs so far did not surface any network issues 🤞 (the only failure was a flaky test on the IT stage)
https://github.com/camunda/zeebe/actions/workflows/ci.yml?query=branch%3Ameg-11590-aws-arm-runners

## Related issues

closes #11590

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

### CODE REVIEW

File: .github/actions/setup-zeebe/action.yml [hunk 1]
```diff
@@ -122,7 +119,7 @@ runs:
         -D maven.artifact.threads=32
         EOF
     - name: Cache local Maven repository
-      if: inputs.java == 'true' && startsWith(runner.name, 'actions-runner-')
+      if: inputs.java == 'true' && (startsWith(runner.name, 'actions-runner-') || startsWith(runner.name, 'aws-') || startsWith(runner.name, 'gcp-'))
```
Comment (megglos) [inline .github/actions/setup-zeebe/action.yml:122]: I don't really like this, do you think it's worth extracting the logic to another job where we could make use of regex and output a simple selfHosted boolean flag to be used in the conditions? 🤔

**Review by npepinpe — APPROVED** [review]
:tada

File: .github/actions/setup-zeebe/action.yml (hunk 1 above)
Comment (megglos) [reply .github/actions/setup-zeebe/action.yml]:
```
```
+ [[ aws-arm-core-8-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ gcp-core-2-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ n1-standard-8-netssd-preempt-quick =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ whatever =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=gh-hosted
result=gh-hosted
```

File: .github/actions/setup-zeebe/action.yml [hunk 2]
```diff
@@ -121,28 +118,29 @@ runs:
...
+      id: runner-env
+      shell: bash
+      # it matters for caching as absolute paths on self-hosted and Github runners differ
+      # self-hosted: `/runner/` vs gh-hosted: `/home/runner`
+      run: |
+        if [[ "${{ runner.name }}" =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]; then
+          echo "result=self-hosted" >> $GITHUB_OUTPUT
+        else
+          echo "result=gh-hosted" >> $GITHUB_OUTPUT
+        fi
```
Comment (megglos) [inline .github/actions/setup-zeebe/action.yml:132]:
```
+ [[ aws-arm-core-8-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ gcp-core-2-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ n1-standard-8-netssd-preempt-quick =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ whatever =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=gh-hosted
result=gh-hosted
```

Comment (megglos) [reply .github/actions/setup-zeebe/action.yml:132]: @npepinpe could you take a look at the new commit?

**Review by npepinpe — APPROVED** [review]
:+1:

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors abort

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors cancel

**Comment by megglos** [comment]
bors merge

**Pull request merged**